### PR TITLE
[WIP] Added more constants

### PR DESCRIPTION
**Pull request by DavidHusicka**
Added more constants

Used JS as reference of missing mathematical constants.

Added constants are:
E, LN2, LN10, LOG2E, LOG10E, SQRT12, SQRT2

## Why?

These constants will be useful for wide range of computations. Also I noticed that some constants are defined but not accessible from GDScript (this makes them all accessible).

I don't expect this to be in 3.2 as all new feature additions are frozen. Let this be part of 4.0.

### CODE REVIEW

**Comment by Chaosus** [comment]
Note, that all of these new constants need to be implemented across the engine including GDNative, VisualScript, and C#, otherwise it is not appropriate for our quality control. Don't hurry, there a lot of time before 4.0 is landed to freeze state.

**Comment by bydavenet** [comment]
> across the engine
Ok, I'll have a look.

**Comment by aaronfranke** [comment]
It would be nice if you could demonstrate that these constants are useful for games. Features with use cases are better than features without use cases.

**Comment by bydavenet** [comment]
Ok, I'll use demonstrate the usage when I make the implementation across the whole engine (I don't have much time lately so it might be pretty late).

File: modules/mono/glue/Managed/Files/Mathf.cs [hunk 1]
```diff
@@ -15,6 +15,14 @@ public static partial class Mathf
...
         public const real_t Inf = real_t.PositiveInfinity;
         public const real_t NaN = real_t.NaN;
+        public const real_t E = (real_t) 2.7182818284590452353602874714M;
+        public const real_t Ln2 = (real_t) 0.6931471805599453094172321214581765680755001343602M;
+        public const real_t Ln10 = (real_t) 2.3025850929940456840179914546843642076011014886287M;
+        public const real_t Log2e = (real_t) 1.4426950408889634073599246810018921374266459541529M;
+        public const real_t Log10e = (real_t) 0.4342944819032518276511289189166050822943970058036M;
+        public const real_t Sqrt12 = (real_t) 0.7071067811865475244008443621048490M;
+        public const real_t Sqrt2 = (real_t) 1.4142135623730950488016887242M;
+
```
Comment (aaronfranke) [inline modules/mono/glue/Managed/Files/Mathf.cs]: You have an extra new line here. More importantly, you need to remove these from `MathfEx.cs` if you add them here.

**Review by aaronfranke — COMMENTED** [review]
Overall, I think E and Sqrt2 are useful, they're also already in C#. Sqrt12 is a variant of Sqrt2 and can be achieved with Sqrt2 / 2 anyway so IMO it's only worth including if it's demonstrated to be very useful.

No clue how useful the log constants are, actually I've never seen Log2e/Log10e used anywhere. These need justification.

While you're at it, we may wish to expose Epsilon. This constant is 1e-6 but becomes 1e-14 if `REAL_T_IS_DOUBLE`.

**Comment by bydavenet** [comment]
> While you're at it, we may wish to expose Epsilon. This constant is 1e-6 but becomes 1e-14 if `REAL_T_IS_DOUBLE`.

Ok, I can add Epsilon but I don't know how to make type checking for constants. You can add this constant to this if you know how. Otherwise I would add two new constants (EPSILON, EPSILON_DOUBLE).

**Comment by Calinou** [comment]
@bydavenet You can probably define the constant conditionally with `#ifdef REAL_T_IS_DOUBLE`.

**Comment by aaronfranke** [comment]
@DavidHusicka  Is this still desired? If so, it needs to be rebased on the latest master branch.

**Comment by DavidHusicka** [comment]
@aaronfranke I am planning to reimplement this again in a new branch and use this as a little reference since I heard there were huge changes in GDScript in the meantime and it might be better to implement this feature from scratch I guess. It is still desired but you can close this PR since there will come new one using the new branch (I just don't know when I will implement this).

**Pull request closed without merging**